browser/base/content/newtab/newTab.js

```diff
@@ -35,6 +35,68 @@ function inPrivateBrowsingMode() {
   return PrivateBrowsingUtils.isContentWindowPrivate(window);
 }
 
+function handleCommand(command, data) {
+  let commandHandled = true;
+  switch(command) {
+    case "NewTab:UpdateTelemetryProbe":
+      Services.telemetry.getHistogramById(data.probe).add(data.value);
+      break;
+    case "NewTab:Register":
+      registerEvent(data.type);
+      break;
+    case "NewTab:GetInitialState":
+      getInitialState();
+      break;
+    default:
+      commandHandled = false;
+  }
+  return commandHandled;
+}
+
+function initRemotePage() {
+  // Messages that the iframe sends the browser will be passed onto
+  // the privileged parent process
+  let iframe = document.getElementById("meep");
+  iframe.addEventListener("load", (e) => {
```

Comment:
This could lead to a race condition, like the iframe could load before init because it is cached. 

Maybe:

```JS
let loaded = (e) => {  ... the stuff in the arrow function...  };
if(contentWindow.document.readyState === "complete"){
   loaded();
   return;
}
iframe.addEventListener("load", loaded)